src/utils/subtestsUtils.ts

```diff
@@ -0,0 +1,114 @@
+import { subtestsOverTimeView } from '../common/constants';
+import {
+  getPerfherderSubtestsCompareWithBaseViewURL,
+  getPerfherderSubtestsCompareOverTimeViewURL,
+} from '../logic/treeherder';
+import type {
+  CompareResultsItem,
+  SubtestsRevisionsHeader,
+} from '../types/state';
+import type { TimeRange } from '../types/types';
+
+type SubtestsResults = {
+  key: string;
+  // By construction, there should be only one item in the array. But if more
+  // than one subtests share the same name, then there will be more than one item.
+  // Can this happen? We're not sure.
+  value: CompareResultsItem[];
+};
+
+export function getSubtestsHeaderAndPerfherderURL(
```

Comment:
There's no need to have just one function for this. I think it would be better to have 2: one for the header, one for the URL. As you'll see in another comment, we shouldn't use the header to compute the URL anyway.